packages/state-transition/src/cache/epochTransitionCache.ts

```diff
@@ -467,12 +467,12 @@ export function beforeProcessEpoch(
       headStakeByIncrement: prevHeadUnslStake,
     },
     currEpochUnslashedTargetStakeByIncrement: currTargetUnslStake,
-    eligibleValidatorIndices,
     indicesToSlash,
     indicesEligibleForActivationQueue,
     indicesEligibleForActivation,
     indicesToEject,
     nextEpochShufflingActiveValidatorIndices,
+    totalNextEpochShufflingActiveIndices,
```

Comment:
can you call this `nextEpochShufflingActiveIndicesLength`?

Reply:
It would be interesting to see how https://github.com/tc39/proposal-resizablearraybuffer performs here

eg:
```ts
// in epochTransitionCache.ts

// top-level declaration
const nextEpochShufflingActiveValidatorIndices = new Uint32Array(new ArrayBuffer(0, {maxByteLength: PRACTICAL_MAX_ACTIVE_INDICES_LENGTH}));

// in beforeProcessEpoch
nextEpochShufflingActiveValidatorIndices.buffer.resize(validatorCount * 4);

/////

// in epochShuffling.ts

export function computeEpochShuffling(
  ...,
  activeIndices: Uint32Array,
  activeIndicesLength: number
) {
  ...
  const _activeIndices = activeIndices.slice(0, activeIndicesLength);
  ...
}
```

Reply:
yes right now the resize is only available for Array not typed array, once it's available we can apply for other places too

Reply:
it appears that resizable ArrayBuffer is in node 20 already so we may begin experimentation